fix(logging) for no-memory log the payload and its size

Tieske opened the pull request.

Tieske committed:
fix(logging) for no-memory log the payload and its size

hbagdi commented on lib/resty/worker/events.lua:
```diff
@@ -155,7 +155,9 @@ local function post_event(source, event, data, unique)
-                "'opts.shm_retries' or increasing the shm size")
+                "'opts.shm_retries'. Payload size: ", #json, " bytes.")
+      log(DEBUG, "worker-events: failed event data, source: '", tostring(source),
+                "', event: '", tostring(event), "', data: ", cjson.encode(data))
```
This can potentially expose sensitive information in logs like credentials, certificates and private keys.

Tieske replied:
thx. fixed, including another occurence.

Tieske committed:
fix(logging) do not log potentially sensitive data

Tieske committed:
change(shm) changed shm_retries default to 999

Effectively it should retry until successful, since we're using
the SHM in a way that it will always be full anyway.
